src/Plugins/RestServer/Helpers/ContractHelper.cs

```diff
@@ -0,0 +1,181 @@
+                    var balanceOfMethod1 = manifest.Abi.GetMethod("balanceOf", 1);
+                    var balanceOfMethod2 = manifest.Abi.GetMethod("balanceOf", 2);
+                    var tokensOfMethod = manifest.Abi.GetMethod("tokensOf", 1);
+                    var ownerOfMethod = manifest.Abi.GetMethod("ownerOf", 1);
+                    var transferMethod1 = manifest.Abi.GetMethod("transfer", 3);
+                    var transferMethod2 = manifest.Abi.GetMethod("transfer", 5);
+
+                    var symbolValid = symbolMethod.Safe == true &&
+                        symbolMethod.ReturnType == ContractParameterType.String;
+                    var decimalsValid = decimalsMethod.Safe == true &&
+                        decimalsMethod.ReturnType == ContractParameterType.Integer;
+                    var totalSupplyValid = totalSupplyMethod.Safe == true &&
+                        totalSupplyMethod.ReturnType == ContractParameterType.Integer;
+                    var balanceOfValid1 = balanceOfMethod1.Safe == true &&
+                        balanceOfMethod1.ReturnType == ContractParameterType.Integer &&
+                        balanceOfMethod1.Parameters[0].Type == ContractParameterType.Hash160;
+                    var balanceOfValid2 = balanceOfMethod2?.Safe == true &&
+                        balanceOfMethod2?.ReturnType == ContractParameterType.Integer &&
+                        balanceOfMethod2?.Parameters[0].Type == ContractParameterType.Hash160 &&
+                        balanceOfMethod2?.Parameters[0].Type == ContractParameterType.ByteArray;
+                    var tokensOfValid = tokensOfMethod.Safe == true &&
+                        tokensOfMethod.ReturnType == ContractParameterType.InteropInterface &&
+                        tokensOfMethod.Parameters[0].Type == ContractParameterType.Hash160;
+                    var ownerOfValid1 = ownerOfMethod.Safe == true &&
+                        ownerOfMethod.ReturnType == ContractParameterType.Hash160 &&
+                        ownerOfMethod.Parameters[0].Type == ContractParameterType.ByteArray;
+                    var ownerOfValid2 = ownerOfMethod.Safe == true &&
+                        ownerOfMethod.ReturnType == ContractParameterType.InteropInterface &&
+                        ownerOfMethod.Parameters[0].Type == ContractParameterType.ByteArray;
+                    var transferValid1 = transferMethod1.Safe == false &&
+                        transferMethod1.ReturnType == ContractParameterType.Boolean &&
+                        transferMethod1.Parameters[0].Type == ContractParameterType.Hash160 &&
+                        transferMethod1.Parameters[1].Type == ContractParameterType.ByteArray &&
+                        transferMethod1.Parameters[2].Type == ContractParameterType.Any;
+                    var transferValid2 = transferMethod2?.Safe == false &&
+                        transferMethod2?.ReturnType == ContractParameterType.Boolean &&
```

Comment:
As well as NEP-11. We have to remove these safe method check, they are not forced in protocol although the latest compiler forced people to do that: https://github.com/neo-project/proposals/blob/master/nep-17.mediawiki

Reply:
But it shows for https://github.com/neo-project/proposals/blob/master/nep-11.mediawiki#totalsupply that it is enforced.

Reply:
Oh, my fault. Yes, should be safe.